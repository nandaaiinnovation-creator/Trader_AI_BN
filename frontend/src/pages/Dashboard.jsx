import React, {useEffect, useState} from 'react'
import { io } from 'socket.io-client'

export default function Dashboard(){
  const [connected, setConnected] = useState(false)
  const [events, setEvents] = useState([])

  useEffect(()=>{
    const socket = io(undefined, { path: '/socket.io' })
    socket.on('connect', ()=> setConnected(true))
    socket.on('disconnect', ()=> setConnected(false))
    socket.on('signal', (data)=>{
      setEvents(e=>[data, ...e].slice(0,50))
    })
    return ()=> socket.disconnect()
  },[])

  return (
    <div>
      <p>Socket: {connected ? 'connected' : 'disconnected'}</p>
      <section>
        <h2>Recent signals</h2>
        <ul>
          {events.map((ev, i)=> <li key={i}>{JSON.stringify(ev)}</li>)}
        </ul>
      </section>
    </div>
  )
}
