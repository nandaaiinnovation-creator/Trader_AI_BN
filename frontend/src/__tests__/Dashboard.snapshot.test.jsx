import React from 'react'
import { render, waitFor } from '@testing-library/react'
import Dashboard from '../pages/Dashboard'

// Provide an explicit mock so imported named functions return Promises
jest.mock('../api/rules', () => ({
  __esModule: true,
  getRuleConfigs: jest.fn().mockResolvedValue({ data: [] }),
  upsertRuleConfig: jest.fn().mockImplementation((name, payload) => Promise.resolve({ data: { name, ...payload } })),
}))
jest.mock('socket.io-client')

test('Dashboard snapshot', async ()=>{
  // ensure api returns no events and no rules quickly
  const { container } = render(<Dashboard />)
  // wait for any async effects to settle
  await waitFor(() => {
    // no-op assertion just to wait for effects
    expect(container).toBeTruthy()
  })
  expect(container).toMatchSnapshot()
})
