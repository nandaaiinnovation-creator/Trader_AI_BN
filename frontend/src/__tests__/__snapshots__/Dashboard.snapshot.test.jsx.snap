// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Dashboard snapshot 1`] = `
<div>
  <div
    class="bn-dashboard"
  >
    <div>
      <header
        class="bn-header"
      >
        <h2
          style="margin: 0px;"
        >
          Signal feed
        </h2>
        <div
          style="font-size: 12px; color: rgb(198, 40, 40);"
        >
          Socket: 
          disconnected
        </div>
      </header>
      <div
        class="bn-card"
        style="margin-top: 12px; position: relative;"
      >
        <div
          class="chart-current-value"
        >
          Current: —
        </div>
        <div
          class="chart-panel"
          style="width: 100%; height: 360px; position: relative;"
        >
          <div
            style="width: 100%; height: 100%;"
          />
          <div
            aria-live="polite"
            class="chart-legend"
            role="status"
            style="position: absolute; right: 8px; top: 8px; background: rgb(255, 255, 255); padding: 4px 8px; border-radius: 4px; box-shadow: 0 1px 3px rgba(0,0,0,0.08); font-size: 12px;"
          >
            Current: —
          </div>
          <div
            aria-hidden="true"
            class="chart-tooltip"
            role="tooltip"
            style="display: none; position: absolute; pointer-events: none; transform: translateX(-50%); background: rgb(34, 34, 34); color: rgb(255, 255, 255); padding: 6px 8px; border-radius: 4px; font-size: 12px;"
          />
        </div>
      </div>
      <section
        class="bn-card"
        style="margin-top: 12px;"
      >
        <h3
          style="margin-top: 0px;"
        >
          Recent signals
        </h3>
        <ul
          class="bn-recent-list"
        />
      </section>
    </div>
    <aside
      class="bn-aside"
    >
      <div
        class="bn-card"
      >
        <div>
          <h3>
            Rules
          </h3>
          <ul />
        </div>
      </div>
    </aside>
  </div>
</div>
`;
